zdns/main.go

```diff
@@ -139,6 +142,37 @@ func main() {
 		gc.NameServers = strings.Split(*servers_string, ",")
 		gc.NameServersSpecified = true
 	}
+	if *localaddr_string != "" {
+		for _, la := range strings.Split(*servers_string, ",") {
+			ip := net.ParseIP(la)
+			if ip != nil {
+				gc.LocalAddrs = append(gc.LocalAddrs, ip)
+				log.Info("using local address: ", localaddr_string)
+			} else {
+				log.Fatal("Invalid argument for --local-addr (", la, "). Must be a comma-separted list of valid IP addresses.")
+			}
+		}
+		gc.LocalAddrSpecified = true
+	}
+	if *localif_string != "" {
+		if gc.LocalAddrSpecified {
+			log.Warn("Both --local-addr and --local-interface specified. Ignoring interface.")
```

Comment:
It might be better to Fatal here rather than Warn. That will make sure that the user knows of the problem. I'm always bit hesitant to eat incoherent configurations rather than let them pass through with a log entry.

Reply:
Changed to fatal and removed the "Ignoring..." part